atomix/cluster/src/main/java/io/atomix/raft/storage/log/RaftLog.java

```diff
@@ -132,16 +133,28 @@ public boolean isEmpty() {
     return journal.isEmpty();
   }
 
-  @SuppressWarnings("unchecked")
   public IndexedRaftLogEntry append(final RaftLogEntry entry) {
-    final byte[] serializedEntry = serializer.serialize(entry);
     final JournalRecord journalRecord;
 
     if (entry.isApplicationEntry()) {
```

Comment:
Do we have a follow up issue to think about getting rid of this if/else/else thing?

Reply:
Not yet. I will create one.